droidlet/tools/hitl/dashboard_app/backend/dashboard_server.py

```diff
@@ -225,5 +229,48 @@ def get_model_value(batch_id, key):
         emit(DASHBOARD_EVENT.GET_MODEL_VALUE.value, [key, get_value_by_key(model, key)])
 
 
+@socketio.on(DASHBOARD_EVENT.GET_MODEL_CHECKSUM.value)
+def get_model_checksum(model_name, agent):
+    """
+    get the checksum for a specific model and agent
+    - input:
+        - model name
+        - agent name
+        - the valid combinations for model name and agent are:
+            - nlu
+            - perception locobot
+            - perception craftassist
+    - output: the checksum if model and agent combination are valid, and if checksum has been computed; otherwise error code
+    """
+    print(
+        f"Request received: {DASHBOARD_EVENT.GET_MODEL_CHECKSUM.value}, model = {model_name}, agent = {agent}"
+    )
+    checksum, error_code = get_model_checksum_by_name_n_agent(model_name, agent)
+    if error_code:
+        emit(DASHBOARD_EVENT.GET_MODEL_CHECKSUM.value, [model_name, agent, error_code])
+    else:
+        emit(DASHBOARD_EVENT.GET_MODEL_CHECKSUM.value, [model_name, agent, checksum])
+
+
+@socketio.on(DASHBOARD_EVENT.UPDATE_MODEL_CHECKSUM.value)
+def update_model_checksum(model_name, agent):
+    """
+    update the checksum for a specific model and agent
```

Comment:
I think we need to be really clear what this does, both in this comment as well as maybe some text description shown to the user.  "Update checksum" implies to me that we're updating it to a value we specify.  This computes the checksum based on the artifacts currently in place on your branch, and then pushes that checksum to the tracked file.